### PR TITLE
feat(text): allow to pass intl message instead of children

### LINKED ISSUE #637: [Idea] Allow Text components to accept an intl message
**Is your feature request related to a problem? Please describe.**
At the moment, we usually wrap the intl messages (e.g. `FormattedMessage`) with the `Text` components.

```js
<Text.Body>
  <FormattedMessage {...messages.title} />
</Text.Body>
```

**Describe the solution you'd like**
Allow to pass the intl message directly to the `Text` component to avoid the wrapping

```js
<Text.Body intlMessage={messages.title} />

// the prop can also be named differently, e.g. `message` but we should probably make it explicit that it relates to react-intl
```

### PR DESCRIPTION
**Pull request by emmenko**
Fixes #637

### CODE REVIEW

File: src/components/typography/text/text.js [hunk 1]
```diff
@@ -63,9 +80,10 @@ Subheadline.propTypes = {
     'positive',
     'negative',
   ]),
-  children: PropTypes.node.isRequired,
   title: nonEmptyString,
   truncate: PropTypes.bool,
+  intlMessage: intlMessageShape,
```
Comment (tdeekens) [inline src/components/typography/text/text.js]: Should this be flipped to the case below. Given there are no children then this?

**Comment by montezume** [comment]
Can you add the new VRT's at the end to make reviewing easier 🙏

File: src/components/typography/text/text.js (hunk 1 above)
Comment (emmenko) [reply src/components/typography/text/text.js]: Ah thanks, good catch!

**Comment by emmenko** [comment]
@montezume do you want to review/approve the VRT?

**Comment by montezume** [comment]
@emmenko 👍 done

File: src/components/typography/text/text.js [hunk 2]
```diff
@@ -26,17 +34,22 @@ const Headline = props => {
...
 };
 
 Headline.displayName = 'TextHeadline';
 Headline.propTypes = {
   elementType: PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired,
-  children: PropTypes.node.isRequired,
   title: nonEmptyString,
   truncate: PropTypes.bool,
+  intlMessage: requiredIf(intlMessageShape, props => !props.children),
+  children: requiredIf(PropTypes.node, props => !props.intlMessage),
```
Comment (tdeekens) [inline src/components/typography/text/text.js]: Thanks!

**Pull request merged**